tests/lib/acpi/src/main.c

```diff
@@ -17,14 +33,44 @@ ZTEST(acpi, test_mcfg_table)
 	zassert_not_null(mcfg, "Failed to get MCFG table");
 }
 
-ZTEST(acpi, test_irq_routing_table)
+ZTEST(acpi, test_dev_enum)
 {
-	static ACPI_PCI_ROUTING_TABLE irq_prt_table[CONFIG_ACPI_MAX_PRT_ENTRY];
-	int status;
+	struct acpi_dev *dev;
+	ACPI_RESOURCE *res_lst;
+	int ret;
+
+	Z_TEST_SKIP_IFNDEF(ACPI_DT_HAS_HID(DT_NODELABEL(pcie0)));
+
+	dev = acpi_device_get(PCIE_HID, PCIE_UID);
+
+	zassert_not_null(dev, "Failed to get acpi device with given HID");
+
+	ret = acpi_current_resource_get(dev->path, &res_lst);
+
+	zassert_ok(ret, "Failed to get current resource setting");
+}
+
+
+ZTEST(acpi, test_resource_enum)
+{
+	struct acpi_dev *dev;
+	struct acpi_irq_resource irq_res;
+	struct acpi_mmio_resource mmio_res;
+	int ret;
+
+	Z_TEST_SKIP_IFNDEF(ACPI_DT_HAS_HID(DT_NODELABEL(rtc0)));
```

Comment:
Are you sure this does the right thing? The `IFNDEF` in the name implies that it's only checking if the given macro is defined or not, however I think the macro will always be defined, either with the value `1` or `0`, so the test would never be skipped.

Instead, you may need to use something like:
```c
	if (!DT_NODE_EXISTS(DT_ALIAS(acpi_dev))) {
		ztest_test_skip();
	}
```

Reply:
see my previous comments